kolibri/plugins/learn/assets/src/state/actions/main.js

```diff
@@ -463,6 +463,7 @@ function showExam(store, id, questionNumber) {
           router.getInstance().replace({ name: PageNames.EXAM_LIST });
           return;
         }
+        exam.closed = true;
```

Comment:
This seems to be setting the exam to closed any time the exam is shown, which doesn't seem quite right.

The issue is arising because the `UserExamResource` has cached the previous call to its endpoint, so is not being updated when we navigate back to the exam list page.

I would suggest clearing this cache, once the exam has been submitted by the student. So, perhaps here, in the `closeExam` action: https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/learn/assets/src/state/actions/main.js#L673
Could be:
```
  return ExamLogResource.getModel(examLog.id)
    .save(examLog)
    .catch(error => {
      handleApiError(store, error);
    });
  return ExamLogResource.getModel(examLog.id)
    .save(examLog)
    .then(UserExamResource.clearCache)
    .catch(error => {
      handleApiError(store, error);
    });
```

This would clear the cache, once the examLog has properly saved, and before the redirect happens.

Reply:
@rtibbles Thanks! Much better approach